startup.py

```diff
@@ -0,0 +1,136 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+import sys
+import re
+import glob
+
+from sgtk.platform import SoftwareLauncher, SoftwareVersion, LaunchInformation
+
+class PhotoshopLauncher(SoftwareLauncher):
+    """
+    Handles the launching of Photoshop. Contains the logic for
+    scanning for installed versions of the software and
+    how to correctly set up a launch environment for the tk-photoshopcc
+    engine.
+    """
+
+    def __init__(self, *args, **kwargs):
+        """
+        Constructor
+        """
+        # init base class
+        super(PhotoshopLauncher, self).__init__(*args, **kwargs)
+        # define our minimum supported version
+        self.minimum_supported_version = "2015.5"
```

Comment:
I'm thinking instead of hardcoding here, maybe we should have a setting in the engine? I know we hardcode a minimum version in engines during `init_engine` as well. Seems like it could be useful to standardize the setting across our engines to prevent duplication and try to keep the value in one place. Similar to the other end of version support with the minimum compatibility setting other engines have. What do you think?